pkg/kubelet/kubelet_node_status.go

```diff
@@ -298,7 +298,12 @@ func (kl *Kubelet) setNodeAddress(node *api.Node) error {
 		if err != nil {
 			return fmt.Errorf("failed to get node address from cloud provider: %v", err)
 		}
-		node.Status.Addresses = nodeAddresses
+		for _, address := range nodeAddresses {
+			if address.Type == api.NodeHostName {
+				return fmt.Errorf("HostnameAddress should not be set by cloudprovider.")
```

Comment:
@liggitt The rationale behind it is: currently the cloud providers do not use do this, we can assume a node to know its hostname, and users specifying a hostname via `--hostname-override` on kubelet invocation do not want the cloud provider to have override this w/ their value (we do not trace whether the user did a hostname override, we consider it initially in `util/node/node.go`).